test/integ/quote-classic.test.ts

```diff
@@ -1987,6 +1990,113 @@ describe('quote', function () {
                 });
               });
             });
+
+            // FOT swap only works for exact in
+            if (type === 'EXACT_INPUT') {
+              const tokenInAndTokenOut = [
+                [BULLET, WETH9[ChainId.MAINNET]!],
+                [WETH9[ChainId.MAINNET]!, BULLET],
+              ];
+
+              tokenInAndTokenOut.forEach(([tokenIn, tokenOut]) => {
+                // FOT integ-test at URA level doesn't need to be as complex as
+                // - at routing-api level (https://github.com/Uniswap/routing-api/blob/ab901a773db4ca31eef0ad731014fc6873c9c6aa/test/mocha/integ/quote.test.ts#L1064-L1242)
+                // - at sor level (https://github.com/Uniswap/smart-order-router/blob/9da29a7f1898e1c09aa1e286a4062919746f04e5/test/integ/routers/alpha-router/alpha-router.integration.test.ts#L2531-L2693)
+                // At URA level, the FOT integ-test just need to ensure after the pass-through flag enableFeeOnTransferFeeFetching gets to routing-api and SOR,
+                // FOT tax gets populated in the tokenIn/tokenOut/tokenInReserve/tokenOutReserve fields in the route as part of the quote response at URA level.
+                it(`fee-on-transfer ${tokenIn.symbol} -> ${tokenOut.symbol}`, async () => {
+                  // we want to swap the tokenIn/tokenOut order so that we can test both sellFeeBps and buyFeeBps for exactIn vs exactOut
+                  const originalAmount = tokenIn.equals(WETH9[ChainId.MAINNET]!) ? '10' : '2924';
+                  const amount = await getAmountFromToken(type, tokenIn, tokenOut, originalAmount);
+                  const simulateFromAddress = tokenIn.equals(WETH9[ChainId.MAINNET]!)
+                    ? '0x2fEb1512183545f48f6b9C5b4EbfCaF49CfCa6F3'
+                    : '0x171d311eAcd2206d21Cb462d661C33F0eddadC03';
```

Comment:
This is a great question.. I was using `alice.address` for `simulateFromAddress`, and the tenderly simulation kept failing with user insufficient balance. And also only `simulateFromAddress` needs to use the whale address, meanwhile recipient can use the `alice.address`, and the calldata with `alice.address` recipient can still get executed on the hardhat mainnet fork successfully. All those things together have puzzled me for a while..

Then I finally realized `simulateFromAddress` actually cannot use the `alice.address`, but to use the token in whale address. The reason is because the tenderly simulation forks from the mainnet itself, as the simulation request from routing-api/SOR. Tenderly fork has no awareness of the hardhat mainnet fork during the test runtime, so `alice.address` on the tenderly fork most likely will have insufficient token in balance, since tenderly fork does not impersonate the whale addresses to fund against `alice.address`.

Then I noticed for all the existing URA integ-test, `simulateFromAddress` all has to explicitly use the token-in whale address:
<img width="1196" alt="Screenshot 2023-10-19 at 10 33 37 AM" src="https://github.com/Uniswap/unified-routing-api/assets/91580504/6d3d283d-ae1e-43cb-9a23-f106547ff4c2">

Reply:
aw as to why two `simulationFromAddress`, because `0x2fEb1512183545f48f6b9C5b4EbfCaF49CfCa6F3` is for WETH whale, and `0x171d311eAcd2206d21Cb462d661C33F0eddadC03` is for BULLET whale.